bootique/src/main/java/io/bootique/config/jackson/InPlaceResourceOverrider.java

```diff
@@ -1,41 +1,36 @@
 package io.bootique.config.jackson;
 
 import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.node.JsonNodeFactory;
-import com.fasterxml.jackson.databind.node.ObjectNode;
 
 import java.net.URL;
-import java.util.List;
 import java.util.Optional;
 import java.util.function.BinaryOperator;
 import java.util.function.Function;
 
 /**
- * Overrides JsonNode object values from one or more configuration resources.
+ * Overrides JsonNode object values from one configuration resource.
  *
  * @since 0.24
  */
 public class InPlaceResourceOverrider implements Function<JsonNode, JsonNode> {
 
-    private List<URL> sources;
+    private URL source;
     private Function<URL, Optional<JsonNode>> parser;
     private BinaryOperator<JsonNode> merger;
 
-    public InPlaceResourceOverrider(List<URL> sources, Function<URL, Optional<JsonNode>> parser, BinaryOperator<JsonNode> merger) {
-        this.sources = sources;
+    public InPlaceResourceOverrider(URL source, Function<URL, Optional<JsonNode>> parser, BinaryOperator<JsonNode> merger) {
+        this.source = source;
         this.parser = parser;
         this.merger = merger;
     }
 
     @Override
     public JsonNode apply(JsonNode jsonNode) {
-        JsonNode configNode = sources.stream()
-                .map(parser::apply)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .reduce(merger)
-                .orElseGet(() -> new ObjectNode(new JsonNodeFactory(true)));
-
-        return merger.apply(jsonNode, configNode);
+        Optional<JsonNode> configNode = parser.apply(source);
```

Comment:
Not sure if it's more readable, but you can use chain functions of `Optional` instead of if checks:

```
        return parser.apply(source)
                .map(node -> merger.apply(jsonNode, node))
                .orElse(jsonNode);
```